### ISSUE TITLE
cocoapods audit and update

### ISSUE THREAD

**Issue by halmueller** [TreeMama/iSeaTree-React-Prototype#280]
We check our Cocoapods sources into our own repo. That makes life much easier for people who don't ordinarily use Xcode but who want to build the project onto their personal iOS devices.

Many of these Cocoapods have had updates. 

The "platform" line targets iOS 10, as does the main project. That's quite old (we're currently at iOS version 15, and Apple is no longer issuing security updates for anything older than iOS 15).

Take a look at these dependencies, remove any that we aren't using, update where available. Change the Cocoapods target version and the main project version to something more current.

**Comment by andaniel2029** [comment]
Hi,  @halmueller 

I checked it.

The current ios target is 10.0.

I updated it to 12.0 and build it on mac successfully.

And in my opinion, we don't need to push the Pods folder and pod. lock file on GitHub repository, because those will be generated by pod install commands like node_modules and package-lock.json file.

About this point, discussed with @TreeMama  on zoom call as well.

Let me push about this ticket.



=======================================


And please check here.

https://github.com/TreeMama/iSeaTree-React-Prototype/commit/ba6c9e650c2d181bde15400213996875657e6813

You were going to read VersionNumber and BuildNumber from the txt file, of course, i understand your idea, but You made some mistakes there.

**1. type error: you wrote '=' two times**
       
       versionCode = = file('../../BuildNumber.txt').text.trim()
       versionName = file('../../VersionString.txt').text.trim()

![image](https://user-images.githubusercontent.com/51064752/151330384-b6844390-2eee-4913-aff6-fbcb3fe9a7ae.png)


**2. when I build it using your codes after fix '=' problem, it doesn't build the Android version successfully still.**
       
         txt file path issue

![image](https://user-images.githubusercontent.com/51064752/151330808-87af62ac-96d0-4fd2-8587-ce2c99d056d6.png)

**3. finally got this error window.**

![image](https://user-images.githubusercontent.com/51064752/151331357-c74fc3a7-aee0-420b-8ae7-c8944b6c25b1.png)

So when I put BuildNumber and VersionString there directly, build success!

![image](https://user-images.githubusercontent.com/51064752/151332526-4a84bbf7-1b7d-48e6-847d-bee9ee008c4a.png)

**Comment by halmueller** [comment]
I disagree with you regarding checking in the Cocoapods code. It's important to keep the project accessible to non-iOS devs who just want to write React/JS. Don't make those folks deal with Cocoapods. 

As for the version numbering: you obviously understand what I was trying to achieve. Can you please make it work? I am beyond exasperated with version numbering, to the point that I am ready to leave the project. This stuff goes out with my name attached. I need it to be consistent between platforms. I want x.y.x(www) to be identical on Android and iOS. Almost every release, this synchronization becomes a minor issue.

**Comment by andaniel2029** [comment]
- Cocoapods modules

I understand you, sometimes that works, but sometimes we need to install it again because all user doesn't have same environment like the laptop performance and specs, Xcode version etc.

In my experience, I had to install it again when I want to run it on another laptop.

Btw, I regard your opinion as well, Let me push updated Cocoapods modules and Podfile.lock file.

- BuildNumber and VersionString issue

I gonna push it and will check more.

**Comment by andaniel2029** [comment]
cocopods updated to 12.0

https://github.com/TreeMama/iSeaTree-React-Prototype/commit/688bb45e4e16ee44560c58351710803ee79e1710

**Comment by halmueller** [comment]
You have not updated the Firebase tools to v8.

**Comment by andaniel2029** [comment]
I am moving this over to this feature branch for future work (sorry about the direct merge). The pod dependencies aren't in this branch (yet) but I agree we should put them in there
